mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -520,29 +544,32 @@ private class LeastSquaresAggregator(
    * Add a new training data to this LeastSquaresAggregator, and update the loss and gradient
    * of the objective function.
    *
-   * @param label The label for this data point.
-   * @param data The features for one data point in dense/sparse vector format to be added
-   *             into this aggregator.
+   * @param instance  The data point instance to be added.
    * @return This LeastSquaresAggregator object.
    */
-  def add(label: Double, data: Vector): this.type = {
-    require(dim == data.size, s"Dimensions mismatch when adding new sample." +
-      s" Expecting $dim but got ${data.size}.")
+  def add(instance: Instance): this.type = instance match {
```

Comment:
Since you already move `case Instance(label, weight, features) =>` to new line, let's do

```scala
def add(instance: Instance): this.type = {
  instance match { case Instance(label, weight, features) =>
  ...
  }
}
```